proposals/0052-fonts.md

```diff
@@ -0,0 +1,471 @@
+**If you have feedback and the feature is released as experimental, please leave it on the Stage 3 PR. Otherwise, comment on the Stage 2 issue (links below).**
+
+- Start Date: 2024-10-15
+- Reference Issues: <!-- related issues, otherwise leave empty -->
+- Implementation PR: https://github.com/withastro/astro/pull/12775
+- Stage 2 Issue: https://github.com/withastro/roadmap/issues/1037
+- Stage 3 PR: https://github.com/withastro/roadmap/pull/1039
+
+# Summary
+
+Have first-party support for fonts in Astro.
+
+# Example
+
+```js
+// astro config
+export default defineConfig({
+  fonts: {
+    families: ["Roboto", "Lato"],
+  },
+});
+```
+
+```astro
+---
+// layouts/Layout.astro
+import { Font } from 'astro:fonts'
+---
+<head>
+	<Font family='Inter' preload />
+	<Font family='Lato' />
+	<style>
+		h1 {
+			font-family: var(--astro-font-inter);
+		}
+		p {
+			font-family: var(--astro-font-lato);
+		}
+	</style>
+</head>
+```
+
+# Background & Motivation
+
+Fonts is one of those basic things when making a website, but also an annoying one to deal with. Should I just use a link to a remote font? Or download it locally? How should I handle preloads then?
+
+The goal is to improve the DX around using fonts in Astro.
+
+> Why not using fontsource?
+
+Fontsource is great! But it's not intuitive to preload, and more importantly, doesn't have all fonts. The goal is to have a more generic API for fonts (eg. you want to use a paid provider like adobe).
+
+# Goals
+
+- Specify what font to use
+- Cache fonts
+- Specify what provider to use
+- Load/preload font on a font basis
+- Generate fallbacks automatically
```

Comment:
is this configurable? should it be?

Reply:
It is not configurable atm. If we can generate one we will and it will take precedence over the fallbacks defined by the user. I guess we could make it configurable, what name do you think would be good? `automaticFallback: false`?